doc/variant2/reference.adoc

```diff
@@ -725,7 +727,8 @@ template<size_t I, class... T>
 [none]
 * {blank}
 +
-Effects: :: If `v.index()` is `I`, returns a reference to the object stored in
+Requires: :: `I < sizeof...(T)`. Otherwise, the program is ill-formed.
```

Comment:
This one isn't true, I think.

Reply:
In your implementation it (rightly) causes a compiler error. This is also what std::variant does: http://eel.is/c++draft/variant.get#3

Reply:
It causes a substitution failure in `variant_alternative_t`. This is equivalent to a compile-time error when there are no other overloads, but not always.

Of course the documentation of `get_if` disagrees so I need to fix that instead.

Reply:
Given that `I` is known statically and that the `sizeof...(T)` is known statically, there shouldn't be any problem with statically rejecting code based on condition `I < sizeof...(T)` in all cases. Or am I missing something?

Reply:
Thanks. In that case, maybe:

```
Remarks: :: This function does not participate in overload resolution unless `I < sizeof...(T)` is `true`.
```